src/components/Tokens/TokenDetails/TokenDetail.tsx

```diff
@@ -174,6 +169,94 @@ const MissingData = styled.div`
   gap: 20px;
 `
 
+export const AboutContainer = styled.div`
+  gap: 16px;
+  padding: 24px 0px;
+`
+
+const TokenDescriptionContainer = styled.div`
+  overflow: hidden;
+  text-overflow: ellipsis;
+  max-width: 100%;
+  max-height: fit-content;
+  padding-top: 16px;
+  line-height: 24px;
+  white-space: pre-wrap;
+`
+
+const TruncateDescriptionButton = styled.div`
+  color: ${({ theme }) => theme.textSecondary};
+  font-weight: 400;
+  font-size: 14px;
+  padding: 14px 0px;
+
+  &:hover,
+  &:focus {
+    color: ${({ theme }) => darken(0.1, theme.textSecondary)};
+    cursor: pointer;
+  }
+`
+
+const TRUNCATE_CHARACTER_COUNT = 400
+
+type TokenDetailData = {
+  description: string | null
+  homepageUrl: string | null
+  twitterName: string | null
+}
+
+const truncateDescription = (desc: string) => {
+  //trim the string to the maximum length
+  let tokenDescriptionTruncated = desc.slice(0, TRUNCATE_CHARACTER_COUNT)
+  //re-trim if we are in the middle of a word
+  tokenDescriptionTruncated = `${tokenDescriptionTruncated.slice(
+    0,
+    Math.min(tokenDescriptionTruncated.length, tokenDescriptionTruncated.lastIndexOf(' '))
+  )}...`
+  return tokenDescriptionTruncated
+}
+
+export function AboutSection({ address, tokenDetailData }: { address: string; tokenDetailData: TokenDetailData }) {
+  const [isDescriptionTruncated, setIsDescriptionTruncated] = useState(true)
+
+  const shouldTruncate =
+    tokenDetailData && tokenDetailData.description
+      ? tokenDetailData.description.length > TRUNCATE_CHARACTER_COUNT
+      : false
+
+  const tokenDescription =
+    tokenDetailData && tokenDetailData.description && shouldTruncate && isDescriptionTruncated
+      ? truncateDescription(tokenDetailData.description)
+      : tokenDetailData.description
+
+  return (
+    <AboutContainer>
+      <AboutHeader>
+        <Trans>About</Trans>
+      </AboutHeader>
+      {(!tokenDetailData || !tokenDetailData.description) && (
+        <NoInfoAvailable>
+          <Trans>No token information available</Trans>
+        </NoInfoAvailable>
+      )}
+      <TokenDescriptionContainer>
+        {tokenDescription}
+        <TruncateDescriptionButton onClick={() => setIsDescriptionTruncated(!isDescriptionTruncated)}>
+          {isDescriptionTruncated ? <Trans>Read more</Trans> : <Trans>Hide</Trans>}
```

Comment:
@infredible should be say "show less" or "collapse" instead of "hide"? it only truncates, doesn't hide.

Reply:
I covered this w/ andy and he preferred the succintness. Appreciate the feedback here tho